### PR TITLE
Integration and Bug Fixes

### LINKED ISSUE #104: Support building forge with a system install of glbinding
Just to be proactive here, I intend to package `forge` to enable the graphics functionalities of `arrayfire` in Debian. Since `forge` now depends on `glbinding`, I also have to package the latter (which is almost done).

Please consider supporting a build of `forge` using a system installation of `glbinding`. [`globjects`](https://github.com/cginternals/globjects/blob/master/source/globjects/CMakeLists.txt) shows how to discover and integrate `glbinding` into a client target.

Cheers,

### PR DESCRIPTION
**Pull request by shehzan10**
* Add ability to use system glbinding. This will allow ArrayFire (among other builders) to build glbinding once and use it for both ArrayFire and Forge.
* Bug fixes in various rendering classes

- Fixes #104

### CODE REVIEW

**Comment by pavanky** [comment]
@shehzan10 should I merge?

File: src/backend/opengl/CMakeLists.txt [hunk 1]
```diff
@@ -186,12 +195,18 @@ TARGET_LINK_LIBRARIES(forge
...
     PRIVATE ${OPENGL_gl_LIBRARY}
-    PRIVATE ${GLBINDING_LIBRARIES}
     PRIVATE ${FREEIMAGE_LIBRARY}
     PRIVATE ${X11_LIBS}
     )
 
-ADD_DEPENDENCIES(forge ${glsl_shader_targets} glbinding)
+ADD_DEPENDENCIES(forge ${glsl_shader_targets})
+IF(USE_SYSTEM_GLBINDING)
+    TARGET_LINK_LIBRARIES(forge PRIVATE glbinding::glbinding)
```
Comment (pavanky) [inline src/backend/opengl/CMakeLists.txt]: Why `glbinding::glbinding` ? I mean what does `::` signify here ?

**Comment by shehzan10** [comment]
I wanted to see if @9prady9 has any comments.
It's not a blocker as the branch is under arrayfire/forge.

File: src/backend/opengl/CMakeLists.txt (hunk 1 above)
Comment (shehzan10) [inline src/backend/opengl/CMakeLists.txt]: I believe thats one way CMake is allowing the libraries to be linked. From what I've read, it handles everything, ie. includes, build options (defines, cxx etc), and libraries.
glbinding link: https://github.com/cginternals/glbinding#tips-for-linking

Comment (ghisvail) [inline src/backend/opengl/CMakeLists.txt]: @pavanky This is how the new style CMake IMPORTED targets work. CMake recommends the usage of a namespace for client-side discovery, hence the `glbinding::` prefix.

**Pull request merged**